src/main/webapp/app/shared/util/utils.tsx

```diff
@@ -176,23 +176,87 @@ export async function isPromiseOk(promise: Promise<any>) {
   }
 }
 
-export function parseAlterationName(name: string) {
-  const regex = new RegExp('([A-Z][0-9]+)([^0-9/]+)/(.+)', 'i');
+// splits alteration name separated by / into multiple alterations
+export function expandAlterationName(name: string) {
+  const regex = new RegExp('^([A-Z])\\s*([0-9]+)\\s*([A-Z])\\s*((?:/\\s*[A-Z]\\s*)*)$', 'i');
   const parts = regex.exec(name);
+
   if (!parts) {
     return [name];
   }
 
-  const alterations = [];
-  const firstPart = parts[1].trim();
-  alterations.push(firstPart + parts[2].trim());
-  const rest = parts[3].split('/');
-  for (const part of rest) {
-    alterations.push(firstPart + part.trim());
+  const alterations: string[] = [];
+  const firstPart = parts[1] + parts[2];
+  alterations.push(firstPart + parts[3]);
+  const rest = parts[4].substring(1);
+  const alleles = rest ? rest.split('/') : [];
+  for (const allele of alleles) {
+    alterations.push(firstPart + allele.trim());
   }
   return alterations;
 }
 
+// splits alteration name into alteration, excluding and comment
+// if alteration is separated by /, applies the same excluding and comment to separated alterations
+export function parseAlterationName(alterationName: string): { alteration: string; excluding: string[]; comment: string; name: string }[] {
+  let regex = new RegExp('\\[(.*)\\]', 'i');
+  const nameSection = regex.exec(alterationName);
+  let name = '';
+  if (nameSection?.length > 1) {
+    name = nameSection[1];
+  }
```

Comment:
What's the logic here?

Reply:
This is to check for the variant name in brackets. I use a regex, and if there is a match, the length should be two. Index 0 will be the whole match, and index 1 will be the captured group (the variant name)

Reply:
oh right, I forgot about the first element is the whole match. Thanks.